### ISSUE TITLE
Non-struct pointer values cannot be used

### ISSUE THREAD

**Issue by MrSimonEmms** [caarlos0/env#235]
When using a non-struct pointer value, we cannot pass in existing data.

```go
package main

import (
	"encoding/json"
	"fmt"
	"log"
	"os"

	"github.com/caarlos0/env/v6"
	"k8s.io/utils/pointer"
)

type Test struct {
//	Bool *bool `env:"TEST"`
	Str *string `env:"TEST"`
}
type ComplexConfig struct {
	Foo Test `envPrefix:"FOO_"`
	Bar Test `envPrefix:"BAR_"`
	D   Test `envPrefix:"D_"`
}

func main() {
	os.Setenv("FOO_TEST", "1")
	os.Setenv("BAR_TEST", "0")

	cfg := ComplexConfig{
		D: Test{
			Str: pointer.String("hello"),
//			Bool: pointer.Bool(true),
		},
	}
	err := env.Parse(&cfg)
	if err != nil {
		log.Fatal(err)
	}
	confBytes, _ := json.Marshal(cfg)
	fmt.Printf("%s", confBytes)
}
```

Result

```bash
env: env: expected a pointer to a Struct
```

It works fine if the `cfg` variable had nothing set for `D` (ie, `cfg := ComplexConfig{}`):

```bash
{"Foo":{"Str":"1"},"Bar":{"Str":"0"},"D":{"Str":null}}
```

The use-case for this is to use it with our [Installer to fill in the configuration](https://github.com/gitpod-io/gitpod/blob/ea45fd8618f7c7449a4aa91472d1ee3d3e68ece6/install/installer/pkg/config/v1/config.go#L216)

**Comment by akutuev** [comment]
@caarlos0 are you going to take a look or can I try to investigate?

**Comment by caarlos0** [comment]
feel free to try, I don't know when I'll have the time to look into it

**Comment by caarlos0** [comment]
fixed in #236

**Comment by MrSimonEmms** [comment]
Thanks @caarlos0
